apache/yunikorn-k8shim
# [YUNIKORN-1650] Add toggle for adding default queue in admission controller

### What is this PR for?
This PR add a configuration to skip adding default queue name label to the pods.

This option is needed when complex placement rules are enabled. Especially when _provided_ rule is used in conjunction with other placement rules and provided rule is higher in the hierarchy. If default queue label is always getting added for pods that do not have a queue label, all the apps will be scheduled via _provided_ rule and the other rules after that will never be executed. 

### What type of PR is it?

### Todos

### What is the Jira issue?
* Open an issue on Jira https://issues.apache.org/jira/browse/YUNIKORN-1650

### How should this be tested?
By default, the configuration is turned off
1. Out of the box, no changes are required if you do not intent to change the current behavior.
2. For skipping to add the default queue name, add this configuration in the admission controller configs: `admissionController.label.defaultQueue=""`. The queue name label should only get added to the pod if a value for queue name was specified during the app submission
3.  For changing the default queue name to something other than `root.default`, add this configuration to the admission controller configs: `admissionController.label.defaultQueue="root.mydefault"`. Note: The value for the default queue has to be a fully qualified queue name.

### Screenshots (if appropriate)

### Questions:
